apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,44 @@
     this.logger.debug('Clock updated', now);
   };
 
+	LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    var self = this;
+		// returns alarm info from DataStore
+		navigator.getDataStores('alarms')
+			.then( function(stores){
+        stores[0].getLength().then(function(len){
+          if(len > 0) {
+            stores[0].get(1)
+              .then( function(data) {
+                if(data.data) {
+                  if(data.data.hour && data.data.minute) {
+                    var type = 'AM',
+                        hour = parseInt(data.data.hour);
+
+                    // Decide wheter to use AM or PM
+                    if(hour > 12) {
+                      type = 'PM';
+                      hour = hour - 12;
+                    }
+
+                    var fullAlarmTime = hour + ':' + data.data.minute + type;
+                    self.resources.elements.alarmtime.textContent =
+                      fullAlarmTime;
+                    self.resources.elements.alarm.classList.remove('no-alarms');
+                    self.logger.debug('Alarm updated to :', fullAlarmTime);
+                    return data;
```

Comment:
If the main purpose of this function is to render the thing, you don't need to return the data.